edb/pgsql/compiler/relctx.py

```diff
@@ -483,13 +483,31 @@ def new_primitive_rvar(
             assert prefix_path_id is not None, 'expected a path'
 
             flipped_id = path_id.extend(ptrref=rptrref)
-            rref = pathctx.get_path_output(
-                set_rvar.query, flipped_id, aspect='identity', env=ctx.env)
+
+            # Unfortunately we can't necessarily just install the
```

Comment:
Hm, I think it might be worth pulling out into its own function for cleanliness and readability but probably not in claiming that it is really any more general than being able to operate on overlay-stacks?

Reply:
Just wondering if this is the only such spot.  But, yeah, extracting into a helper seems reasonable even if there's only one call site.